### PR TITLE
MINT-5961 Migrate to SQLAlchemy version 2

### PR DESCRIPTION
**Pull request by atemp21**
JIRA-Ticket with summary line

https://hingehealth.atlassian.net/browse/MINT-5961

### Purpose
- Migrate to SQLAlchemy version 2

### CODE REVIEW

File: app/alembic/env.py [hunk 1]
```diff
@@ -102,8 +102,8 @@ def run_migrations_online():
             compare_type=True,
             version_table_schema=target_schema,
         )
-        connection.execute(f"CREATE SCHEMA IF NOT EXISTS {target_schema}")
-        connection.execute(f"SET SEARCH_PATH TO {target_schema}")
+        connection.execute(text(f"CREATE SCHEMA IF NOT EXISTS {target_schema}"))
```
Comment (atemp21) [inline app/alembic/env.py:105]: direct string SQL no longer supported: use `text()`

File: app/app/models/patient.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
-import sqlalchemy as sa  # type: ignore
+from typing import Optional, Any
+from sqlalchemy.orm import Mapped, mapped_column
 
 from app.db.base_class import Base
 from app.db.types import phi_column as phi
 
 
 class Patient(Base):
```
Comment (atemp21) [inline app/app/models/patient.py:8]: declarative mapping style/syntax changed in version 2
https://docs.sqlalchemy.org/en/20/orm/mapping_styles.html#declarative-mapping

File: app/app/crud/base.py [hunk 3]
```diff
@@ -24,12 +25,12 @@ def __init__(self, model: Type[ModelType]):
         self.model = model
 
     def get(self, db: Session, id: Any) -> Optional[ModelType]:
-        return db.query(self.model).filter(self.model.id == id).first()
```
Comment (atemp21) [inline app/app/crud/base.py:27]: `db.query()` deprecated. use `select` now 
https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-20-query-usage

**Review by gyandeeps — APPROVED** [review]

**Review by ninjawuguay — APPROVED** [review]

**Review by ak47 — APPROVED** [review]
Outstanding! Thank you for blazing the trail

**Pull request merged**